### ISSUE TITLE
Updated Documentation Request - dashboard and PEP exclusions

### ISSUE THREAD

**Issue by kript** [irods/irods_rule_engine_plugin_audit_amqp#87]
I have a memory that before a certain version of the plugin, one needed to exclude the PEP's that uploaded data as the contents of the file was included with it - if my memory is correct, can the README be updated with the caveat for the version please?

Also, in the demo, there was a dashboard, could the demo dashboard be made available in the repo as a starting point for people to edit please? [this might be it?](https://github.com/irods/irods_demo/blob/main/audit/entrypoint.sh)

cheers

John

**Comment by kript** [comment]
Also, its is not clear from the example configuration how one points at a remote ElasticSearch instance.

I presume you change `amqp_location`, something like ;

```
"amqp_location" : "<user>@<Elastic server>:5672",
```
but doesn't that infer that the ElasticSearch config needs to be running an AMQP system, which I thought the plugin installed itself? 

Apologies if I am missing something obvious...

I would also appreciate any notes on using an central AMQP server, as opposed to the current approach which seems to be installing amqp on sever iRODS server that wants to audit activity?

Thanks!

**Comment by trel** [comment]
Elastic doesn't speak AMQP directly.

The demo installs a message broker (ActiveMQ (or RabbitMQ)), and then logstash pulls the messages from the broker and pushes them into elastic.  So, the `amqp_location` will be a message broker of your choice.

**Comment by trel** [comment]
This is the example kibana dashboard used in the training:
   https://github.com/irods/irods_training/blob/main/advanced/example_kibana_dashboard.json

**Comment by trel** [comment]
not sending ALL the file contents was fixed in 4.2.11.0 ... https://github.com/irods/irods_rule_engine_plugin_audit_amqp/issues/61

**Comment by trel** [comment]
1) You can upgrade :P

Or...

2) cherry-pick https://github.com/irods/irods_rule_engine_plugin_audit_amqp/commit/d65e133bfe09f489e53e40a0c12e513fcde929b2 and run an engineering build until upgrading to 4.2.11.0.

Or...

3) cherry-pick and turn into a proper 4.2.7.2 release.

---


Probably option 2 is the fastest, then 3 (depends on us), then 1 (depends on lots).

**Comment by kript** [comment]
(3) would definitely be my preference. How long would that be? 🙏

**Comment by kript** [comment]
In the meantime, can we exclude the PEP?

```
        {
            "instance_name": "irods_rule_engine_plugin-audit_amqp-instance",
            "plugin_name": "irods_rule_engine_plugin-audit_amqp",
            "plugin_specific_configuration" : {
                 "amqp_location" : "localhost:5672",
                 "amqp_options" : "",
                 "amqp_topic" : "audit_messages",
                 "pep_regex_to_match" : "(^audit_pep_api_((?!gen_query|auth_response|auth_request|**data_obj_write**).)*_(pre|post)$|audit_pep_api_auth_.*_except)"
                 }
            },
```
If so, are there any other PEP's we should exclude to avoid tickling the bug (like cp, rsync)?

**Comment by trel** [comment]
Yes, excluding the PEP is a valid workaround.    Issue #61 was originally found due to `audit_pep_api_data_obj_put_pre`.

`cp` doesn't receive the buffer from the client.  `rsync`... would on a local-to-iRODS transfer?

**Comment by trel** [comment]
re option 3...   candidate PR for 4.2.7.2 at https://github.com/irods/irods_rule_engine_plugin_audit_amqp/pull/89

**Comment by trel** [comment]
engineering build sent.
